### ISSUE TITLE
FTP sender initial directory

### ISSUE THREAD

**Issue by danielfaust** [mendhak/gpslogger#564]
I had an issue with a custom FTP server (Python) where it wasn't able to store the uploaded files because the directory name in the settings was empty (testfile upload failed).

I fixed this by explicitly entering "/" into the directory field inside GPSLogger's FTP settings (nothing else, because it stores into the root directory of the account).

I'm not sure if having the directory being empty is the default and if it is normal for FTP servers not to expect an initial directory, but I'm opening this issue because it may be one. If I remember correctly, I've always had to set an initial directory containing at a minimum the slash it in FTP clients.

**Comment by mendhak** [comment]
The default is a folder called 'gpslogger' (needs to be added in place of the blank folder name) - during the test upload did the dialog box show the error and the FTP commands/responses?  

Normally when the gpslogger folder doesn't exist the code [tries to create it](https://github.com/mendhak/gpslogger/blob/master/gpslogger/src/main/java/com/mendhak/gpslogger/senders/ftp/FtpJob.java#L145)

**Comment by danielfaust** [comment]
Ok, that commit fixes it.

My FTP server does not allow the creation of directories, maybe I didn't read the error details properly as to notice that it wasn't attempting to upload to the root diretory but to GPSLogger instead.

Now that I removed the slash, it is still working, so uploading to root is not a problem if a slash is missing.

It's the best solution to add the `android:defaultValue`, but I'm a bit concerned about that lowercasing of the directory. It does not affect me, but it could cause problems for those who rely on the default, unless of course the default is persisted after the first application load right after a fresh install (I don't know if that is the case). I mean because of case sensitivity on unix-like OSs and possibly some Windows FTP-Servers.

Thanks for your effort.

**Comment by mendhak** [comment]
OK That's fixed. 

For testing FTP I normally use [this docker image](https://hub.docker.com/r/stilliard/pure-ftpd/~/dockerfile/)

**Comment by danielfaust** [comment]
I think I found my problem.

I'm using non-standard ports.

The first time I entered the port, I guess I entered a wrong one. When pressing the back button to get out of the long-waiting connection attempt, then changing the port, and then retrying the upload, the port was still the previous one, not the updated one.

There is some issue with the propagation of the port value when closing the "port number input dialog".

It's as if it gets applied later, after the erroneous connection attempt times out. As if there were now two connection attempts and the second one can't succeed until the first one times out, which happens completely in the background, as it doesn't have an UI anymore, since I had pressed Android's back button to back out of the connection attempt.

I still don't really understand what's going on there, I'll take a look at the code when I have some time, but atm I don't have much of it for troubleshooting stuff that doesn't affect me directly.

It's mainly an annoyance to/bug that affects new users, and that only if they start using FTP, which I believe won't be many.
